users/resources.py

```diff
@@ -104,6 +104,33 @@ def delete(self, request, pk):
         return Response(status=status.HTTP_204_NO_CONTENT)
 
 
+class StudentPrizesResource(APIView):
+    serializer_class = PrizeSerializer
+    permission_classes = [permissions.IsAuthenticated]
+
+    def get(self, request, pk):
+        user_id = request.user.id
+
+        if not has_user_access_to_student(user_id, pk):
+            raise PermissionDenied
+
+        student = Student.objects.get(pk=pk)
+        prizes = Prize.objects.filter(student=student)
```

Comment:
It's not necessary to retrieve student object to later use it in the filtering. Filtering can be done directly using the id value
```suggestion
        prizes = Prize.objects.filter(student=pk)
```

or 

```suggestion
        prizes = Prize.objects.filter(student_id=pk)
```

I believe both options should work